### DISCUSSION TITLE
Tab Closing Event

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [microsoft/terminal#17988] [General]
Is there a .NET event I can use to watch for when a tab is closing?

**Comment by lhecker** [comment]
Do you mean from inside a console application? In that case, you'd do something like this: https://stackoverflow.com/a/474743

**Reply by jdhitsolutions** [reply]
Yes. Ideally, I wanted to create an event subscriber in my PowerShell session. I was hoping that clicking the X on the tab to dismiss it (Ctrl+F4) would fire the event.

**Reply by DHowett** [reply]
It should fire a standard console closing event, which you should be able to catch as documented above. 🙂
